scala-thoughts.md

```diff
@@ -0,0 +1,41 @@
+
+Scala is an incredibly powerful language that I really enjoyed using through 
+these exercises. Although I have worked with quite a bit of Scala code before
+this assignment, I haven't had the opportunity to write much of it from 
+scratch. Additionally, I haven't worked with Scala much since completing PLs
+last semester. Some of the "quirky" nature of Scala makes much more sense after
+having implemented a type checker and made extensive use of Haskell in CS131.
+
```

Comment:
I agree about the Haskell part. Without Haskell, I think a lot of the Scala would have felt much more confusing. I thought it was interesting to see how you approached some of the questions differently (more elegantly) than I did -- maybe a result of your Scala experience.